Makefile

```diff
@@ -132,6 +133,21 @@ else
 GOBIN=$(shell go env GOBIN)
 endif
 
+# Get latest git hash
+GIT_COMMIT_HASH=$(shell git rev-parse HEAD)
+
+# Get the local image registry required for clusterctl upgrade tests
+LOCAL_IMAGE_REGISTRY ?= localhost:5000
```

Comment:
@jimmidyson Yes, we really need image registry that will be available from test Nutanix cluster. 
To test CAPX controller upgrade I've reused clusterctl-upgrade test from CAPI: https://github.com/kubernetes-sigs/cluster-api/blob/v1.3.5/test/e2e/clusterctl_upgrade.go#L209 This test creates bootstrap cluster (kind) next it creates workload cluster on bootstrap and use `cluster init` to make it (workload cluster) the management cluster with old CAPX provider version, then on the mgmt cluster test creates workload cluster. So in fact we have 3 clusters: 1. bootstrap(kind) => 2. management => 3. workload. The management cluster during upgrade requires the latest (e2e) image build which it can not take from ko.local (this cluster is not local kind cluster, rather the cluster on Nutanix test cluster).
In CAPI they recommend to use AWS S3 and docker save to make latest build available for management cluster: https://github.com/kubernetes-sigs/cluster-api/blob/v1.3.5/test/e2e/clusterctl_upgrade.go#L136 But I don't like this approach because it means patch to manifests with additional preKubeadmCommands. So, to solve it I utilize the local harbor using above variable.

Reply:
That makes sense, thanks for explaining.

Reply:
Thanks for the explanation! Makes sense to me!